packages/wrangler/templates/startDevWorker/ProxyWorker.ts

```diff
@@ -124,8 +124,10 @@ export class ProxyWorker implements DurableObject {
 			Object.assign(userWorkerUrl, proxyData.userWorkerUrl);
 
 			// set request.url in the UserWorker
-			const innerUrl = new URL(request.url);
-			Object.assign(innerUrl, proxyData.userWorkerInnerUrlOverrides);
+			const innerUrl = urlFromParts(
+				proxyData.userWorkerInnerUrlOverrides ?? {},
+				request.url
+			);
```

Comment:
This is a refactor to make clearer what is going on in one step rather than two. No change in behaviour.